client/azure/azure.go

```diff
@@ -0,0 +1,132 @@
+package azure
+
+import (
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/Azure/azure-sdk-for-go/arm/resources/resources"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+// Config contains the common attributes to create an Azure client.
+type Config struct {
+	// Dependencies
+	Logger micrologger.Logger
+
+	// ClientID is the ID of the Active Directory Service Principal.
+	ClientID string
+	// ClientSecret is the secret of the Active Directory Service Principal.
+	ClientSecret string
+	// SubscriptionID is the ID of the Azure subscription.
+	SubscriptionID string
+	// TenantID is the ID of the Active Directory tenant.
+	TenantID string
+}
+
+// DefaultConfig provides a default configuration to create an Azure client by
+// best effort.
+func DefaultConfig() *Config {
+	var err error
+
+	var newLogger micrologger.Logger
+	{
+		config := micrologger.DefaultConfig()
+		newLogger, err = micrologger.New(config)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	return &Config{
+		// Dependencies.
+		Logger: newLogger,
+
+		// Settings.
+		ClientID:       "",
+		ClientSecret:   "",
+		SubscriptionID: "",
+		TenantID:       "",
+	}
+}
+
+// Clients is the collection of Azure API clients.
+type Clients struct {
```

Comment:
nitpick: That could be called a `ClientSet` like it is in K8s API. `NewClients` looks to me like it should return a slice.